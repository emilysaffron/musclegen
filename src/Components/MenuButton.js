import styled from "@emotion/styled";
import { Link } from "@reach/router";

const StyledButton = styled.button`
  border-radius: 20px;
  background: #a8bbce;
  width: 250px;
  height: 150px;
  padding: 20px;
  margin: 10px;
  font-size: 50px;
  font-family: Palatino, URW Palladio L, serif;
  color: black;
  cursor: pointer;
  outline: none;
`;

const DropDownButton = styled.button`
  background: none;
  width: max-content;
  font-size: 20px;
  /* See other comment on assigning a font-family at a higher level in the hierarchy to prevent needing to do this */
  font-family: Palatino, URW Palladio L, serif;
  /* Similarly if black is what you want as the default font color, this could be assigned to the html { } selector */
  color: black;
  cursor: pointer;
  outline: none;
  border: none;
`;

const MenuButton = ({ label, dropdown }) => {
  const labelWithoutWhitespace = label.replace(/\s+/g, "-");

  return (
    <Link to={`${labelWithoutWhitespace.toLowerCase()}`}>
      {dropdown ? (
        <DropDownButton>{label}</DropDownButton>
      ) : (
        <StyledButton>{label}</StyledButton>
      )}
    </Link>
  );
};

export default MenuButton;
