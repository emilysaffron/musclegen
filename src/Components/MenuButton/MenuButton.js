import styled from "@emotion/styled";
import { Link } from "@reach/router";

const StyledButton = styled.button`
  border-radius: 20px;
  background: #a8bbce;
  width: 250px;
  height: 150px;
  padding: 20px;
  margin: 10px;
  font-size: 50px;
  color: black;
  cursor: pointer;
  outline: none;
`;

const DropDownButton = styled.button`
  background: none;
  width: max-content;
  font-size: 20px;
  cursor: pointer;
  outline: none;
  border: none;
`;

const MenuButton = ({ label, dropdown }) => {
  const labelWithoutWhitespace = label.replace(/\s+/g, "-");

  return (
    <Link to={`${labelWithoutWhitespace.toLowerCase()}`}>
      {dropdown ? (
        <DropDownButton>{label}</DropDownButton>
      ) : (
        <StyledButton>{label}</StyledButton>
      )}
    </Link>
  );
};

export default MenuButton;
