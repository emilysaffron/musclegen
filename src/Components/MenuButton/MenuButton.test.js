import renderer from "react-test-renderer";
import MenuButton from "./MenuButton";

it("MenuButton Snapshot", () => {
  const tree = renderer
    .create(<MenuButton option="option" dropdown={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
