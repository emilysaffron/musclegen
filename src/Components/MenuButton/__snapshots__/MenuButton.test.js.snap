// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton Snapshot 1`] = `
<a
  href="/option"
  onClick={[Function]}
>
  <button
    className="css-gip7ia"
  >
    option
  </button>
</a>
`;
